src/expr/impl/src/scalar/case.rs

```diff
@@ -106,6 +107,152 @@ impl Expression for CaseExpression {
     }
 }
 
+/// With large scale of simple form match arms in case-when expression,
+/// we could optimize the `CaseExpression` to `ConstantLookupExpression`,
+/// which could significantly facilitate the evaluation of case-when.
+#[derive(Debug)]
+struct ConstantLookupExpression {
+    return_type: DataType,
+    arms: HashMap<ScalarImpl, BoxedExpression>,
+    fallback: Option<BoxedExpression>,
+    const_expression: Option<BoxedExpression>,
```

Comment:
I think the `const operand` optimization could be handled with an optimizer Rule instead of within the executor if we want to make this expression clearer.

Reply:
This optimization could be done in a separated PR.

Reply:
Then I'll just remove this part at present.